File: sdk-go/wave/pagination.go
package wave

// Pagination mirrors the gateway's pagination block.
type Pagination struct {
	Page       int64 `json:"page"`
	PerPage    int64 `json:"perPage"`
	Total      int64 `json:"total"`
	TotalPages int64 `json:"totalPages"`
}

// Page is the standard paginated envelope: {data:[...],pagination:{...}}.
type Page[T any] struct {
	Data       []T        `json:"data"`
	Pagination Pagination `json:"pagination"`
}

File: sdk-rust/wave-core/src/pagination.rs
use serde::{Deserialize, Serialize};

/// Pagination block returned alongside list responses.
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Pagination {
    #[serde(default)]
    pub page: i64,
    #[serde(rename = "perPage", default)]
    pub per_page: i64,
    #[serde(default)]
    pub total: i64,
    #[serde(rename = "totalPages", default)]
    pub total_pages: i64,
}

/// Standard paginated envelope: `{ "data": [...], "pagination": {...} }`.
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Page<T> {
    pub data: Vec<T>,
    #[serde(default = "default_pagination")]
    pub pagination: Pagination,
}

fn default_pagination() -> Pagination {
    Pagination { page: 0, per_page: 0, total: 0, total_pages: 0 }
}
